packages/govuk-frontend/src/govuk/all.mjs

```diff
@@ -119,3 +119,17 @@ export {
  *
  * @typedef {keyof Config} ConfigKey
  */
+
+/**
+ * Schema for components
+ *
+ * @typedef {object} Schema
+ * @property {SchemaItem[]} [anyOf] - An array of conditions
+ */
+
+/**
+ * Component schema fields
+ *
+ * @typedef {object} SchemaItem
+ * @property {string[]} required - Whether the current item is required
+ */
```

Comment:
We may want to keep these types in the base class if we consider them private or internal

Works great where they are but aware they'll go into the JSDoc HTML output:

[/docs/javascript/interfaces/**Schema.html**](https://govuk-frontend-pr-4176.herokuapp.com/docs/javascript/interfaces/Schema.html)
[/docs/javascript/interfaces/**SchemaItem.html**](https://govuk-frontend-pr-4176.herokuapp.com/docs/javascript/interfaces/SchemaItem.html)

**Tip:** To purposely hide things that TypeDoc has discovered we can use **typedoc.config.js** like:

https://github.com/alphagov/govuk-frontend/blob/1d1fb0a535062128556a669b7ae391e509f030d6/packages/govuk-frontend-review/typedoc.config.js#L29-L30